### PR TITLE
Onboarding: Enable token-auth'ed GET settings endpoint

### PR DESCRIPTION
**Pull request by ockham**
WIP. Companion PR to https://github.com/Automattic/wp-calypso/pull/21013.

#### Changes proposed in this Pull Request:

* Allow access to `GET` `/settings` endpoint with an onboarding token.

#### Testing instructions:

* See https://github.com/Automattic/wp-calypso/pull/21013

#### Proposed changelog entry for your changes:

TBD

### CODE REVIEW

**Comment by tyxla** [comment]
Seems like we should flatten the query args for the GET request here (as flattened in https://github.com/Automattic/wp-calypso/pull/21013/commits/d4632bfd58da2a59ca9cf1230f461a4d07f89697)? Or should we unflatten them and update the JP server to support arrays in GET?

File: class.jetpack.php [hunk 1]
```diff
@@ -5312,6 +5312,9 @@ function verify_xml_rpc_signature() {
 		// Let's see if this is onboarding. In such case, use user token type and the provided user id.
 		if ( isset( $this->HTTP_RAW_POST_DATA ) ) {
 			$jpo = json_decode( $this->HTTP_RAW_POST_DATA );
+		} else {
+			$jpo = json_decode( json_encode( $_GET ), FALSE );
```
Comment (eliorivero) [inline class.jetpack.php]: Let's use the standard WP function `wp_json_encode`
https://developer.wordpress.org/reference/functions/wp_json_encode/

**Comment by tyxla** [comment]
I've done some work in #8461 and D9099-code to allow non-flat query args.

**Comment by ockham** [comment]
Closing in favor of #8461.

**Pull request closed without merging**